src/Microsoft.Diagnostics.Monitoring/DiagnosticServices.cs

```diff
@@ -68,6 +69,34 @@ public async Task<Stream> GetDump(int pid, DumpType mode)
             return new AutoDeleteFileStream(dumpFilePath);
         }
 
+        public async Task<Stream> GetGcDump(int pid, TimeSpan timeout, CancellationToken cancellationToken)
+        {
+            using (CancellationTokenSource linkedSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken))
+            {
+                linkedSource.CancelAfter(timeout);
+
+                var graph = new MemoryGraph(50_000);
+                var processor = new DiagnosticsEventPipeProcessor(_contextConfiguration,
```

Comment:
The event pipe processor bakes in a duration. Why pass an infinite duration to it while creating an external timeout?

Reply:
Duration is "collect whatever you can within this amount of time." Timeout is "cancel if you don't complete within this amount of time." For a GC dump, duration is not appropriate but the timeout concept is. The pipe processor doesn't understand the difference between the two concepts and I don't know if I can easily differentiate that from with the HandleGCEvents method.